spec/controllers/pusher_controller_spec.rb

```diff
@@ -0,0 +1,16 @@
+describe PusherController do
+
+  let(:user) { create(:user) }
+  let(:pusher) { create(:pusher) }
+
+  describe 'auth' do
+    context 'when the user is logged in' do
+      before { sign_in user }
+
+    it 'renders pusher response notification' do
+        get :auth
+        expect(response).to render_template("auth")
```

Comment:
What's the error message you're getting when you run this test?

Reply:
`1) PusherController auth when the user is logged in renders pusher response notification
     Failure/Error: response = Pusher[params[:channel_name]].authenticate(params[:socket_id])

     NoMethodError:
       undefined method `length' for nil:NilClass`
`# ./spec/controllers/pusher_controller_spec.rb:11:in `block (4 levels) in <top (required)>'`

Reply:
So you'll want to test two cases:
1) When the user is not signed in
2) When the user is signed in

For case 1) you have an expect that does something like this: `expect(response).to render_template(text: I18n.t('pusher.not_authorized'))`.

For case 2), notice that the response is constructed like so: `response = Pusher[params[:channel_name]].authenticate(params[:socket_id])`

This means you'll need to pass in params for the `channel_name` and `socket_id`.

Let me know if you have any further questions!